CenterForOpenScience/osf.io
# [ENG-4336] [ENG-4337] [ENG-4338] bugfixes and tests for osf.metadata (guid-metadata followup)

## Purpose
add tests (and fix bugs) for `osf.metadata`


## Changes
- add/fill `osf_tests/metadata/`
- fix: `user.social['profileWebsites']` is a list, not a str
- fix: datacite needs >0 creators always
- code cleanup, comments, typehints


## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify metadata view/edit/download works
  - ...for addon files
  - ...for users with profile-websites
- Verify datacite doi minting is unaffected

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Documentation



## Side Effects



## Ticket
[ENG-4336] >0 datacite:creators for addon files
[ENG-4337] unit tests
[ENG-4338] social-personal link serialization



[ENG-4336]: https://openscience.atlassian.net/browse/ENG-4336
[ENG-4337]: https://openscience.atlassian.net/browse/ENG-4337
[ENG-4338]: https://openscience.atlassian.net/browse/ENG-4338
